docs/design/lacp/README.md

```diff
@@ -0,0 +1,77 @@
+# Link Aggregation Control Protocol
+
+## Description
+
+Link Aggregation allows parallel full duplex point-to-point links to be used as if they were a single link. This design is focused on the following scenario.
+
+![LACP usage scenario](lacp-usage-scenario.png "LACP usage scenario")  
+
+LACP aggregator is connected to network controlled by open-kilda using one or multiple links (using one or multiple switches). From open-kilda perspective all ports connected to the same LACP aggregator should be treated as aggregateable links. Kilda should be able to:
+ * establish LACP session with aggregators
+ * create flows terminated on aggregated ports
+ 
+ Traffic from any source endpoint of such flows should be balanced over all destination endpoints which has proper LACP session established.
+
+## LACP session control
+
+![LACPDU capture](lacpdu-capture.png "LACPDU capture")
+
+Additional default openflow rule is installed on each switch to catch LACP packets and send them to the controller. This rule has match by eth_type=0x8809 (Slow protocols).
+
+Separate storm topology is used to handle all LACP related events. Main part of aggregation topology is the AggregatorBolt. Aggregator bolt should use in-memory storage to save information about partner systems connected to different ports. Storage structure: switch-port tuple is a key and partner system info is a value. Partner system info includes system priority, system, key, port priority, port and state. Two links can be aggregated correctly if they have the same partner system and key but different port. Open-kilda can aggregate any ports so it should respond with constant system and key but different port. Port value should be copied from incoming LACPDU to avoid port collisions on partner side. 
```

Comment:
what if topology is restarted, how this lacp sessions should be restored? any side effects on that?

Reply:
topology was removed from the design